### PR TITLE
[syncd/FlexCounter]:Fix last remove bug

### PR DESCRIPTION
**Pull request by Pterosaur**
- The bug description
When the last object in FlexCounter was removed, the FlexCounter is also removed from FlexCounterManager. At the next time, when a new observed object is added and a new FlexCounter will be created. But the new FlexCounter cannot get the configuration. Because the configuration message from FlexCounterDB has been consumed at the first time.

- How to fix it
Add a flag to indicate whether the configuration also be removed from the FlexCounterDB. If not, don't delete the FlexCounter from FlexCounterManager even no observed objects in the FlexCounter.

Signed-off-by: Ze Gan <ganze718@gmail.com>

### CODE REVIEW

**Comment by Pterosaur** [comment]
retest vs please

**Comment by lguohan** [comment]
@qiluo-msft , can you check why this pr is broken?

**Review by lguohan — DISMISSED** [review]

File: syncd/FlexCounter.cpp [hunk 1]
```diff
@@ -26,6 +26,7 @@ FlexCounter::FlexCounter(
     SWSS_LOG_ENTER();
 
     m_enable = false;
+    m_is_discard = false;
```
Comment (kcudnik) [inline syncd/FlexCounter.cpp:29]: please rename to m_isDiscard to follow naming convention

**Comment by qiluo-msft** [comment]
retest vs please

**Comment by qiluo-msft** [comment]
The build should be fixed by https://github.com/Azure/sonic-sairedis/pull/681

**Review by kcudnik — APPROVED** [review]

**Pull request merged**